### ISSUE TITLE
CMI Vault and Towny Bank Balance

### ISSUE THREAD

**Issue by ItsRainingHP** [Zrips/CMI#7081]
### Description of Issue

Towny bank balances do not store in MySQL database. Major issue for many servers. Without this feature I will have to use something other than CMI and I wasted my money on it.

### Version Information

```shell
CMI 9.2.2.0
```

### Errors

```shell
No error logs
```

### Relevant Config Sections

```yaml
storage:
  # storage method, can be MySQL or sqlite
  # ATTENTION! DON'T USE SAME DATABASE TABLES FOR MORE THEN ONE SERVER, YOU WILL HAVE UNEXPECTED ISSUES. EACH SERVER SHOULD HAVE ITS OWN DATABASE TABLE SETUP
  method: MySQL
mysql:
  # Requires Mysql.
  username: CMI
  password: pass
  hostname: IP:3306
  database: CMI
  # Valid characters are a-z 0-9 and _
  tablePrefix: CMI_
  # If you have no clue what these values do, then keep it at default values
  autoReconnect: true
  useSSL: true
  verifyServerCertificate: false
```

### Relevant Plugins

CMI Vault 1.7.3

### Agreements

- [X] My server is supported by CMI.
- [X] My version of CMI at the time of this report is up to date.
- [X] I have searched the github and asked around before making this report.

**Comment by ItsRainingHP** [comment]
For anyone reading this. I decided to right my own plugin to resolve this issues. Running on 1.19 with CMI no issues.

https://www.spigotmc.org/resources/networkbanksync-towny-vault-economy-velocity-mysql.103635/

**Comment by CloudeLecaw** [comment]
nice idea :)

i dont knowy you thought that cmi should save the bank or econmy of an other plugin like towny...but never mind about.. your solution for this is good.

**Comment by ItsRainingHP** [comment]
Well I tried using CMI to do just that but the MySQL table does not have any Town balances or Nation balances. Something I needed for my server. I am currently working on 2.0 version of the plugin which will use Sockets to all Velocity servers on event. Rather than requires the sync interval to trigger.

**Comment by ItsRainingHP** [comment]
Just an update for those that come to read this. I have updated my plugin to support Velocity, Bungeecord, and Waterfall proxy servers. Using sockets, server owners can sync town bank balances across-server networks. Tested on 1.19.1 using CMI and EssentialsX.

NetworkBankSync: https://www.spigotmc.org/resources/networkbanksync-towny-vault-economy-mysql-velocity.103635/
VelocityBankSync: https://www.spigotmc.org/resources/velocitybanksync-networkbanksync-addon-mysql-websockets-velocity.103921/
BungeeBankSync: https://www.spigotmc.org/resources/bungeebanksync-networkbanksync-addon-mysql-websockets-bungeecord-waterfall.103999/
